src/lib.rs

```diff
@@ -462,6 +462,52 @@ impl Waker {
     }
 }
 
+/// Set non-blocking mode on a stream.
+///
+/// This is a convenience function if the source of your stream doesn't provide an
+/// easy way to set it into non-blocking mode.
+///
+/// ## Example
+///
+/// ```
+/// use std::process;
+/// use popol::set_nonblocking;
+///
+/// let child = process::Command::new("ls")
+///     .stdout(process::Stdio::piped())
+///     .spawn()
+///     .unwrap();
+/// let out = child.stdout.unwrap();
+///
+/// set_nonblocking(&out, true).unwrap();
+/// ```
+///
+/// ## Return
+///
+/// On Linux, this should always return `Ok(0)` or `Err(_)`. On other operating systems,
+/// consult the `fcntl(2)` man page.
+pub fn set_nonblocking(fd: &dyn AsRawFd, nonblocking: bool) -> io::Result<i32> {
```

Comment:
I have this returning `i32`, but maybe it should be `libc::c_int`? I figured `i32` was clearer for somebody coming along and reading the docs, but maybe not.